uefi-mkconfig

```diff
@@ -198,58 +157,109 @@ main () {
 
 	# Verify kernel commands
         for check_config in $kernel_commands; do
-
                 # Strip initrd=* from config file
                 [[ "$check_config" != *"initrd="* ]] && strip_kernel_commands="$strip_kernel_commands $check_config"
                 # Verify if "root=" is present
                 [[ "$check_config" == "root="* ]] && valid_kernel_commands=1
-
         done
 	kernel_commands="$strip_kernel_commands"
 
         [[ -z $valid_kernel_commands ]] && ewarn "Warning, kernel command \"root=\" is missing from loaded configuration!"
+	
+	# Get initial state of UEFI entries
+	initial_uefi_state="$(efibootmgr -u)"
+	
+	# Wipe all managed entries for regeneration
+	wipe_entries
+	
+	partition_mounts="${mounted_efi_partitions// "$efi_parttype"/}"
+	
+	# Find all .efi files on this partition and split them to old and new
+	for prefix in $kernel_prefixes; do
+		find_results_new="$(find $partition_mounts -name "$prefix*.efi" ! -name "$prefix*-old.efi" -printf "%f\n")"
+		find_results_old="$(find $partition_mounts -name "$prefix*-old.efi" -printf "%f\n")"
+		[[ -n "$find_results_new" ]] && kernel_images_new+="$find_results_new "
+		[[ -n "$find_results_old" ]] && kernel_images_old+="$find_results_old "
+	done
+
+	# Sort kernel images
+	kernel_images="$(sort -uV <<< "$kernel_images_old") $(sort -uV <<< "$kernel_images_new")"
+
+	# Add path back to the kernel image
+	for kernel_image in $kernel_images; do
+		# Variable partition_mounts in this case can't be inside of ""
+        	partition_efis+="$(find $partition_mounts -name "$kernel_image") "
```

Comment:
Here too, we should add `local partition_efis=` at the beginning

Reply:
I think here is the origin of why this still duplicates entries. I propose to drop the double for loop and simply sort with the full paths. End result is then sorted by mount point and then version, instead of by version and then mount point but I don't think that is a huge issue right?

Reply:
We were able to find the cause. It was caused by spacesat the end of the line on 181 and 182